tap_github/streams.py

```diff
@@ -375,3 +375,112 @@ def post_process(self, row: dict, context: Optional[dict] = None) -> dict:
             ),
         ),
     ).to_dict()
+
+
+class CommitsStream(GitHubStream):
+    """
+    Defines the 'Commits' stream.
+    The stream is fetched per repository to maximize optimize for API quota
+    usage.
+    """
+
+    name = "commits"
+    path = "/repos/{org}/{repo}/commits"
+    primary_keys = ["node_id"]
```

Comment:
I haven't referenced the API docs yet but I would expect this to be "commit_ref" or "sha". "Node" or "Node ID" doesn't seem very intuitive and it looks like perhaps those terms are overloaded several times in the stream properties - what do you think?

Is "node" self-explanatory enough in your opinion or should we alias this or perhaps use another property?

Reply:
The `node_id` is an id that github uses on all of their data elements (users, commits, issues, repos...). It seems to be globally unique, which is why I've used this as the primary key here. I haven't actually found any documentation about this, but looking at a bunch of API call results has led me to think the above.
Theoretically, the SHA is not guaranteed to be unique (see [this SO answer, under note 1](https://stackoverflow.com/a/25128596/1138710)) but I guess the risk of clashes is ridiculously low. So if you think it makes more sense, I don't mind changing here.

Reply:
That's okay. Given your explanation, I think `node_id` makes sense here. I wasn't aware of this design around their API but given that hash collisions do _at least in theory_ happen, I propose keeping `node_id` as you have it.

Reply:
Found this which was helpful and confirming: https://docs.github.com/en/graphql/guides/using-global-node-ids